cluster/engine.go

```diff
@@ -192,6 +196,27 @@ func (e *Engine) Status() string {
 	return "Unhealthy"
 }
 
+// IncFailureCount increases engine's failure count, and set engine as unhealthy if threshold is crossed
+func (e *Engine) IncFailureCount() {
+	e.Lock()
+	e.failureCount++
+	if e.healthy && e.failureCount >= engineFailureCountThreshold {
+		e.healthy = false
+	}
+	e.Unlock()
+}
+
+// SetEngineHealth sets engine healthy state
+func (e *Engine) SetEngineHealth(state bool) {
```

Comment:
This should map with `IsHealthy` to look more like a getter/setter.

Perhaps `IsHealthy()` and `SetHealthy()`?